### PR TITLE
🚧 target Jenkins v164.1

### PR DESCRIPTION
**Pull request by casz**
Here is our checklist for contributors. No hard requirement here, just a reminder

- [x] Please describe what you did

Let's discuss targeting the v2.164.1 LTS, the project is pretty stable.
There are a few PRs we could get in and release before bumping.
The bump would be welcomed this is, after all, JCasC we should be able to safely bump core version without too much fuss from Jenkins users.
Would be great to know we can run on Java 11 runtime!

- [ ] Link to issue you're working on if there's a relevant one

- [x] Did you provide a test-case to demonstrate feature actually works / issue is fixed ?

- [ ] Please also consider adding a line to [CHANGELOG.md](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/CHANGELOG.md)

Changelog will be added later.

### CODE REVIEW

File: plugin/src/main/java/io/jenkins/plugins/casc/core/AdminWhitelistRuleConfigurator.java [hunk 1]
```diff
@@ -56,8 +56,13 @@ public String getName() {
 
     @Override
     protected AdminWhitelistRule instance(Mapping mapping, ConfigurationContext context) {
-        Injector injector = Jenkins.getInstance().getInjector();
-        return injector.getInstance(AdminWhitelistRule.class);
+        Injector injector = Jenkins.get().getInjector();
```
Comment (casz) [inline plugin/src/main/java/io/jenkins/plugins/casc/core/AdminWhitelistRuleConfigurator.java]: I wonder why injector is marked `@CheckForNull`, is there a better way of getting the `AdminWhitelistRule` instance?

**Comment by casz** [comment]
CI is clearly bugged... :cry:

**Comment by casz** [comment]
I honestly have given up on Jenkins CI, Travis at moment does exactly the same as what Jenkins CI does and faster :cry:

**Comment by casz** [comment]
Spotbugs failure on Jenkins CI is due to https://github.com/spotbugs/spotbugs/issues/756

**Comment by jonbrohauge** [comment]
:+1: for :rocket:
We try to stay on the latest stable Jenkins + plugins, and as such we have a stable platform, so I see no reason not to bump to a newer required LTS for this plugin. There is obviously some maintenance, but it's at a minimum, and 99.9% of the maintenance is due to a plugin changing some logic we use.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+language: java
```
Comment (timja) [inline .travis.yml]: not really related to this PR?

Comment (casz) [reply .travis.yml]: Well, it is kind of related. I was blocked by Jenkins CI trying to use JDK 11, and I needed to know if we could build with JDK 11!

**Comment by er1c** [comment]
I just did a lot of work on: https://github.com/jenkinsci/github-plugin/pull/209 would that then cause an issue to to get https://github.com/jenkinsci/github-plugin/pull/210 working?

**Comment by casz** [comment]
There should be no internal changes that should make your work obsolete. You can still use an older version for the plugin testing.

I think one of the main reason I want to bump is to avoid maintaining a lot of legacy/version matching dance to add some functionality like #788 

Perhaps that is a lousy argument but an argument none the less.

Looking at our user installs by Jenkins version is currently not possible.
[INFRA-2004](https://issues.jenkins-ci.org/browse/INFRA-2004)

**Comment by timja** [comment]
Nice it does exist 🎉

**Comment by casz** [comment]
Reading those numbers, it might make sense to do the version dance. 😢

**Comment by jonbrohauge** [comment]
@casz At some point we need to an upgrade of the target Jenkins, might as well be now, i.e. next release. Although v2.164.1 might be too far this time around, if you look at the [stats](https://stats.jenkins.io/pluginversions/configuration-as-code.html).

**Comment by timja** [comment]
Might be best to wait for next months numbers run and see how 2.164.1 is looking

**Comment by casz** [comment]
We can just pin the PR for later and do the version dance in #788

**Comment by oleg-nenashev** [comment]
> Would be great to know we can run on Java 11 runtime!

We can. It was a part of our smoke tests since October

**Comment by casz** [comment]
it was already marked a WIP :cold_sweat: so no worries @oleg-nenashev

**Comment by oleg-nenashev** [comment]
Yeah, just wanted to make sure we discuss it before merge

**Comment by casz** [comment]
Closing as we will most likely split into configurator API

**Comment by 0384223** [comment]
No me lo puedo creer que haya echo algo bien

**Comment by 0384223** [comment]
Gracias por ayudarme a todos y a vosotros en particular gracoas

**Review by Pero110290 — COMMENTED** [review]
Per

**Pull request closed without merging**